contracts/ERC721M.sol

```diff
@@ -397,7 +408,9 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
                     minter,
                     qty,
                     _cosigner,
-                    timestamp
+                    timestamp,
+                    _chainID(),
+                    getCosignNonce(minter)
```

Comment:
double check if burn token will decrease the mintedCount mapping from erc721a

Reply:
EIP712 supports https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator

which means that if a contract has more than 1 domain that needs signature, we might want to add the domain into the hashing seeds.

but since this signature is only used for minting, the default domain is 'minting'